### PR TITLE
[VIT-6695] Pull preferences: ios support

### PR DESCRIPTION
**Pull request by ItachiEU**

### CODE REVIEW

File: Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift [hunk 1]
```diff
@@ -1,4 +1,37 @@
 public enum VitalResource: Equatable, Hashable, Codable {
+  public func resourceToBackfillType() -> BackfillType {
```
Comment (andersio) [inline Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:2]:
```suggestion
  @_spi(VitalSDKInternals)
  public func resourceToBackfillType() -> BackfillType {
```

File: Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift [hunk 2]
```diff
@@ -136,3 +169,33 @@ public enum VitalResource: Equatable, Hashable, Codable {
     }
   }
 }
+
+public enum BackfillType: String, Codable {
```
Comment (andersio) [inline Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:173]:
```suggestion
@_spi(VitalSDKInternals)
public enum BackfillType: String, Codable {
```

Comment (andersio) [reply Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:173]: I would define this as a struct:

```
public struct BackfillType: String, RawRepresentable, Codable {
  public let rawValue: String

  public init(rawValue: String) {
    self.rawValue = rawValue
  }

  public static let workouts = BackfillType(rawValue: "workouts")
  public static let body = BackfillType(rawValue: "body")
  // ...
}
```

Enum is exhaustive so if the sync state endpoint starts returning new BackfillTypes, decoding is going to fail.
(unless we make the sync state endpoint conditionalize on the SDK version header...)

Comment (ItachiEU) [reply Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:173]: so in the case of the struct, in `resourceToBackfillType` mapping we would do:
```swift
default:
      return BackfillType.rawValue;
```
?

Comment (andersio) [reply Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:173]: That shouldn’t need a default case. VitalResource is also exhaustive but the SDK is always built from source, so you won’t have a new case going into an old codepath.

File: Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift (hunk 1 above)
Comment (ItachiEU) [reply Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:2]: The @_spi prevent me from accessing this in `historicalStartDate` function

Comment (ItachiEU) [reply Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:2]: Similarly the @_spi and BackfillType struct causes issue in the model definitions

Comment (andersio) [reply Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift:2]: You will need to update the import in the function’s file with `@_spi(VitalSDKInternals) import VitalCore`

**Review by andersio — APPROVED** [review]

**Comment by andersio** [comment]
Thanks for working this through

**Pull request merged**